### ISSUE TITLE
🐞 Bug report: Sticky Header does not work on resize using FlexBox Containers in templates and main stage

### ISSUE THREAD

**Issue by bowmandesign** [elementor/elementor#18744]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest development version of Elementor.

### Description

Make a header template in the Elementor Theme Builder using Flexbox Containers
Make your Header Sticky Top, animations optional, happens with and without.
Resize your screen, < tablet breaks, main stage disappears.

### Steps to reproduce

Make a header template in the Elementor Theme Builder using Flexbox Containers
Make your Header Sticky Top, animations optional, happens with and without.
Resize your screen, < tablet breaks, stage disappears.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro) with development versions.
- [X] This bug happens with a vanilla WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

*

**Comment by BINPAKISTAN** [comment]
Thanks

**Comment by MichaelDarko** [comment]
Same problem.

**Comment by ijcarson1** [comment]
Bump. Still experiencing this on the latest releases.

**Comment by SaltedLevity** [comment]
I'm experiencing this not just in the theme builder but ANY item/element in a container, including a container. Setting anything to sticky and scrolling down makes the width either shrink to the item size, or grow to the whole viewport width.
I.e. 

1. Container* > heading
Container set to sticky at top, when you scroll the container expands to 100vw (when native width is roughly 500px)
2. Container > heading*
Heading set to sticky at top, when you scroll the heading box size shrinks to just the text's width size, and no longer is the full width of the container.

**Comment by Jixart** [comment]
Any update on this ? anyone found a temporary solution ? i have the same issue here, when sticky is active container will increase size.

**Comment by tauseefch** [comment]
Having same issue when I make any thing sticky in column its width automatically get increased when scrolling

**Comment by mapsmarketing** [comment]
Same here

**Comment by jithinio** [comment]
Same here

**Comment by bluantinoo** [comment]
Same here, really upsetting. Hope will be fixed soon. Untill then I will stick on old sections/columns layout for header

**Comment by mllemata** [comment]
Excuse me dear elementor team, but I am very unhappy client, which is trying to help also everyone in this issue. I have the same problem as well and I've created support ticket, but still have not been resolved because Elementor 1st line support is still annoying with identifying our account, which is active and they know our ID. I hope they finally answer with appropriate technical answer with solution. I dont know whether you have such a bad experience with support like me... but I beg Elementor Support team to provide helpful "tip" quickly without any further question about account...

**Comment by Pixlkraft** [comment]
Have the same issue on all sites using Flexbox containers and sticky header. With all respect, this is a bug that needs a patch quite soon. C'mon, Elementor!!

**Comment by headerlocation** [comment]
Auch ich habe diese Problem. Wird es demnächst behoben oder Gibt es einen Hotfix?

**Comment by estalavera** [comment]
Bug still occurring in the latest version  - 3.11.2

Create a flexbox container then:

- Advanced
1. Sticky: Top
2. Sticky On: Desktop
3. Update the page

Scroll down and the sticky container should work.
The bug occurs when the user resizes the browser window to mobile and back to desktop.

**Comment by TIIUNDER** [comment]
Is there some workaround to fix this?

**Comment by Pixlkraft** [comment]
> Is there some workaround to fix this?

Yes. The width of the outer container/section is calculated through JS. This is the problem with this bug - it doesn't recalculate the width when browser viewport is resized. But you can overwrite the width on the container with css using:

`width: 100% !important`

**Comment by jim28100** [comment]
Pixlkraft - You're a star. Your CSS worked a treat.

This has been driving me crazy. 
I can't believe Elementor hasn't fixed this yet not that Flexbox is a Release Candidate?

**Comment by robertosiimoes** [comment]
Hi everyone,

I use the fix of @Pixlkraft and use it in production websites everything works perfectly : 

`selector.elementor-sticky {
    width: 100% !important;
    left: 0 !important;
}`

With this CSS code solve the problem for example for a Header in sticky mode.
Rewrite it for your use ! 
👌🏼

Best regards.

**Comment by MountainLaurelDigital** [comment]
@Pixlkraft  Thanks Pixel craft. Such a simple solution. I wouldn't have thought !important could override the inline JS that sizes it. So happy this fixes it.

**Comment by thedigitalnick** [comment]
Elementor Pro 3.25 in 2024 and this is still a bug. The fix still works. Thanks @Pixlkraft

**Comment by TigerToni** [comment]
> > Is there some workaround to fix this?
> 
> Yes. The width of the outer container/section is calculated through JS. This is the problem with this bug - it doesn't recalculate the width when browser viewport is resized. But you can overwrite the width on the container with css using:
> 
> `.elementor-sticky { width: 100% !important }`

love you for this!

**Comment by w3programs** [comment]
@Pixlkraft Thank you so much. You're awesome!
